core/esmf-aspect-model-aas-generator/src/main/java/org/eclipse/esmf/aspectmodel/aas/AspectModelAasVisitor.java

```diff
@@ -545,6 +556,8 @@ private <T extends Collection> Environment visitCollectionProperty( final T coll
       final SubmodelElementBuilder builder = property ->
             new DefaultSubmodelElementList.Builder()
                   .idShort( property.getName() )
+                  .semanticId( buildReferenceForCollection( collection.getAspectModelUrn().isPresent()
```

Comment:
If the collection has no URN it is only locally valid for the Property that refers to it. In that case we can't use the collection's local name a semantic ID; instead we need to either (1) add no semantic ID at all for this element (probably the cleanest solution) or (2) create an identifier for the semantic ID that is based on the Property's URN, e.g. by concatenating `propertyName + "Characteristic"`.

Reply:
Yes, thanks for the clarification. I think, 2nd option is more acceptable, because in the task means, that some semantic Id has to be provided.

Thanks!

Reply:
Hi: I also prefer option (1) as @atextor proposed. We should not add IDs that are not existing. 
When writing the task I  forgot to consider the anoymous characteristics. SemanticId is not mandatory so it is fine to not add a semanticId in these cases. (However, it would be best practice to not use anonymous characteristic for collections if aasx should be generated out of it)